src/testResultSummary/index.ts

```diff
@@ -0,0 +1,27 @@
+import * as core from "@actions/core"
+import * as fs from "fs"
+import { TestResultsFileMetadata, SummarizedTestResults } from "./types"
+import { parseGoTestResults } from "./parsers/golang"
+
+/**
+ * Get the test results data from the file and prepare it for loki
+ */
+export function getTestResultSummary(
+  fileMetadata: TestResultsFileMetadata,
+): SummarizedTestResults {
+  try {
+    const fileData = fs.readFileSync(fileMetadata.filePath, "utf8")
+
+    // format the data based on the test type, go is the first but solidity will be added soon
+    switch (fileMetadata.testType) {
+      case "go":
+        return parseGoTestResults(fileData)
+    }
+  } catch (error) {
+    core.warning("Could not read the file: " + fileMetadata.filePath)
+    core.warning("ignoring and moving on.")
+  }
+
+  // if we get here, we failed to handle all the cases
+  throw Error("Could not get test results data")
```

Comment:
We either need to handle the default case above and throw a warning or error message letting people know they used a non supported test type or add that messaging here.

Reply:
I think it'd be good to change this error message to be more specific on the types of tests we support, and which test they supplied, did you want to take this on?

Reply:
We can give users more descriptive error messages by using this zod method: https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md#flattening-errors

Along with a custom error map if need be